### DISCUSSION TITLE
Device config save

### DISCUSSION THREAD

**Discussion by kaizen13** [ipspace/netlab#1047] [General]
Hi,

Sorry for the newbie question but is there an option to save the config i.e wr mem and after netlab down it is retained? Each time I do netlab up && netlab down it seems the config I add is gone. Is the only option: to do netlab collect and then copy paste from the config subfolder?  

I am using clab ; eos and cumulus as recommended in the BGP lab docs.

Thanks.
Cheers,
Emil

**Comment by ipspace** [comment]
First of all, thanks for the question! It's always good to know how others use netlab and what we've overlooked.

For the moment, you have a few options (plus cut-and-paste ;), none of them good:

* Use "netlab clab package" to add pointers to EOS startup configs to clab.yml file. It was designed to create a standalone package that you could send to someone else (or use as a standalone lab), that's why it creates a tarball (.tar.gz) file. I could easily modify it to create just the modified clab.yml file.
* Rename "config" directory to "somedirectory" (not literally, use any name you like ;), rename .cfg files to .j2 files, and run "netlab config somedirectory" to replay the config files to the network devices. Should you go down this path, I'd start the lab with "netlab up --no-config" to skip the initial device configuration and avoid any conflicts (because "netlab config" does not do "config replace"). I could tweak the "netlab collect" playbook and "netlab up" command to make the process a bit more streamlined.
* What I'm doing is to take the previous step, but then clean up the collected config files to contain just the extra changes I want to keep, and then use "netlab config somedirectory"

Finally, keep in mind that this works well for devices that have a single configuration file, but applies only FRR part of configuration to Cumulus Linux or FRR. In theory, we could replace the interface configs on Cumulus Linux as well, but there's no way to do that on FRR as the _netlab_ configuration scripts use Linux commands (not "ifupdown" config file) to configure interfaces.

I'll definitely add something along the line of the above musings to _netlab_ to make this job easier, but it probably won't get into 1.8 release.

Hope this helps,
Ivan

**Reply by kaizen13** [reply]
Thanks Ivan for the help. I will test those options :)

**Reply by ipspace** [reply]
If was simpler than I thought (and the afternoon coffee helped). I'm almost done with 'netlab up --reload-config' ;) Wait a bit for 1.8 release.

**Comment by ipspace** [comment]
@kaizen13 Release 1.8.0 is out. Relevant documentation is in https://netlab.tools/netlab/up/#reloading-saved-configurations, hope it will work for you.
